File: common/interfaces/message-info.ts
import { ChatEmoji } from '../constants/chat-emojis';
import { Message } from './message';
import { UserIdName } from './user-id-name';

export interface MessageInfo {
    roomCode: string;
    message: Message;
}

export interface MessageEmojiInfo {
    messageId: string;
    chatEmoji: ChatEmoji;
    userIdName: UserIdName;
    roomCode?: string;
}

export interface ChatStateInfo {
    roomCode: string;
    playerUsername: string;
}

File: client-mobile/app/src/main/java/com/example/polyquiz/chat/domain/MessageInfo.kt
package com.example.polyquiz.chat.domain
import com.example.polyquiz.auth.domain.UserIdName

data class MessageInfo(
    val roomCode: String,
    val message: Message
)

data class MessageEmojiInfo(
    val messageId: String,
    val chatEmoji: String,
    val userIdName: UserIdName,
    val roomCode: String?
)

data class ChatStateInfo(
    val roomCode: String,
    val playerUsername: String
)
